poc/vdaf_poc/test_utils.py

```diff
@@ -69,18 +71,88 @@ def run_vdaf_test(
         self.assertEqual(agg_result, expected_agg_result)
 
 
+class VdafTestVectorOperationDict(TypedDict):
+    """
+    A description of one of the operations that should be performed with this
+    test vector.
+
+    Attributes:
+        operation: The type of operation to be performed. This is one of
+            "shard", "prep_init", "prep_shares_to_prep", "prep_next",
+            "aggregate", or "unshard".
+        round: The round number of the operation to be performed. This
+            determines which prepare share, prepare state, and/or prepare
+            message to use.
+        aggregator_id: The aggregator ID to use when performing this
+            operation. This determines which messages and which prepare state
+            to use, in addition to the aggregator ID argument itself.
+        report_index: The index of the report on which to perform this
+            operation. This is an index into the `prep` array.
+        success: If this is true, the operation should succeed, and its output
+            should match the corresponding values in the test vector. If this
+            is false, the operation should fail.
```

Comment:
What about cases where the operation succeeds but the output is not the same? I'm thinking of "aggregate" for instance where a bad implementation of `agg_update()` could result in an incorrect aggregate share

Reply:
I think this is covered, by "and its output should match..."